src/rubyfmt.rb

```diff
@@ -2238,6 +2246,38 @@ def format_while(ps, rest)
   ps.emit_newline if ps.start_of_line.last
 end
 
+def format_for(ps, rest)
+  loop_vars, iterable, expressions = rest
+
+  unless Array === loop_vars[0]
+    loop_vars = [loop_vars]
+  end
+
+  ps.emit_indent if ps.start_of_line.last
+
+  ps.emit_for
+  ps.emit_ident(" ")
+  format_list_like_thing_items(ps, [loop_vars], true)
+  ps.emit_ident(" ")
+  ps.emit_in
+  ps.emit_ident(" ")
+  ps.with_start_of_line(false) do
```

Comment:
Not sure this is necessary, but it seemed appropriate to follow the same structure as while loops.

Reply:
if you've ever emitted tokens and not a newline, it is good practice to throw one of these blocks in.